web-ng/src/app/components/feature-panel/feature-panel.component.ts

```diff
@@ -20,31 +20,42 @@ import { FeatureService } from './../../services/feature/feature.service';
 import { map, switchMap } from 'rxjs/operators';
 import { ProjectService } from './../../services/project/project.service';
 import { List } from 'immutable';
-import { Observable } from 'rxjs';
-import { Component } from '@angular/core';
+import { Observable, Subscription } from 'rxjs';
+import { Component, NgZone, OnDestroy, OnInit } from '@angular/core';
 import { Layer } from '../../shared/models/layer.model';
 import { Field, FieldType } from '../../shared/models/form/field.model';
 import { NavigationService } from '../../services/router/router.service';
 import { FeatureHeaderActionType } from '../feature-panel-header/feature-panel-header.component';
+import { ConfirmationDialogComponent } from '../confirmation-dialog/confirmation-dialog.component';
+import { Router } from '@angular/router';
+import { DataStoreService } from '../../services/data-store/data-store.service';
+import { MatDialog } from '@angular/material/dialog';
 
 // TODO: Rename "FeatureDetailsComponent".
 @Component({
   selector: 'ground-feature-panel',
   templateUrl: './feature-panel.component.html',
   styleUrls: ['./feature-panel.component.css'],
 })
-export class FeaturePanelComponent {
+export class FeaturePanelComponent implements OnInit, OnDestroy {
+  projectId?: string | null;
+  observationId?: string | null;
```

Comment:
Can `| null` be removed from these two, letting `?` (undefined) cover the 'missing' case?